### PR TITLE
fix: encode image urls

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@raisedadead this seems to have satisfied CodeQL.

**Review by raisedadead — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
Should be good to go.

File: client/src/components/settings/about.tsx [hunk 1]
```diff
@@ -152,7 +152,7 @@ class AboutSettings extends Component<AboutProps, AboutState> {
     const value = (e.target as HTMLInputElement).value.slice(0);
     // eslint-disable-next-line @typescript-eslint/naming-convention
     if (isURL(value, { require_protocol: true })) {
-      this.validationImage.src = value;
+      this.validationImage.src = encodeURI(value);
```
[the remark this answers is not in the document]
Comment (nhcarrigan) [reply client/src/components/settings/about.tsx:155]: AFAIK, the only place we display the URL is in the settings page. 

image

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/src/components/settings/about.tsx:155]: We shouldn't have to, no.  `value` is just a url entered by the user, so either it's valid and encoding shouldn't matter or it isn't and encoding shouldn't hurt.

**Pull request merged**